File: app-mvvm/src/main/java/com/xuie/imaginaryandroid/gui/nets/news/NewsListContract.java
package com.xuie.imaginaryandroid.gui.nets.news;

import com.xuie.imaginaryandroid.BasePresenter;
import com.xuie.imaginaryandroid.BaseView;
import com.xuie.imaginaryandroid.data.NetsSummary;

import java.util.List;

/**
 * This specifies the contract between the view and the presenter.
 *
 * @author xuie
 */
public interface NewsListContract {

    interface View extends BaseView {

        void addList(boolean isRefresh, List<NetsSummary> netsSummaries);
    }

    interface Presenter extends BasePresenter {
        void getList(boolean isRefresh);
    }
}

File: app-mvp-kotlin/src/main/java/com/xuie/imaginaryandroid/gui/nets/news/NewsListContract.kt
package com.xuie.imaginaryandroid.gui.nets.news

import com.xuie.imaginaryandroid.BasePresenter
import com.xuie.imaginaryandroid.BaseView
import com.xuie.imaginaryandroid.data.NetsSummary

/**
 * This specifies the contract between the view and the presenter.
 */
interface NewsListContract {

    interface View : BaseView<Presenter> {

        fun addList(isRefresh: Boolean, netsSummaries: List<NetsSummary>)
    }

    interface Presenter : BasePresenter {
        fun getList(isRefresh: Boolean)
    }
}
